### PR TITLE
feat: add hertzsentry feature

### PR DESCRIPTION
**Pull request by gityh2021**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,119 @@
-# hertzsentry
\ No newline at end of file
+# hertzsentry
```
Comment (Skyenought) [inline README.md]: 这个pr的名称写的正式一点，你要是觉得在开发中可以写成
- [WIP] feat: add hertzsentry feature

Comment (Skyenought) [inline README.md]: 这个标题：
- hertzsentry -> (This is a community driven project)

File: README.md [hunk 2]
```diff
@@ -1 +1,119 @@
...
+
```
Comment (Skyenought) [inline README.md:2]: 没有介绍基本的关于这个中间件的信息，并不是所有人都知道 sentry 干什么的

File: README.md (hunk 1 above)
Comment (gityh2021) [reply README.md]: OK

File: README.md (hunk 2 above)
Comment (gityh2021) [reply README.md:2]: 👌

**Comment by li-jin-gou** [comment]
- improve pr title / pr desc / readme + resolve ci
- why not use go mod and add  extra dir?

**Comment by gityh2021** [comment]
> * improve pr title / pr desc / readme + resolve ci
> * why not use go mod and add  extra dir?

what does ci mean, I just know CICD.
I find that the dictionary of ginsentry and fibersentry did not contains go mod, so I didn't commit is.
add  extra dir: I write a main file in main package, maybe didn't need it?

**Comment by li-jin-gou** [comment]
> > * improve pr title / pr desc / readme + resolve ci
> > * why not use go mod and add  extra dir?
> 
> what does ci mean, I just know CICD. I find that the dictionary of ginsentry and fibersentry did not contains go mod, so I didn't commit is. add extra dir: I write a main file in main package, maybe didn't need it?

- you need refer to https://github.com/hertz-contrib/keyauth
- you can refer to https://docs.github.com/en/actions/automating-builds-and-tests/about-continuous-integration  about what is ci.

File: README.md [hunk 3]
```diff
@@ -1 +1,123 @@
-# hertzsentry
\ No newline at end of file
+# hertzsentry -> (This is a community driven project)
```
Comment (li-jin-gou) [inline README.md:1]: readme  need to improve

File: README.md [hunk 4]
```diff
@@ -1 +1,123 @@
...
+
+Sentry is an open source real-time error monitoring project that supports many sides, including the Web front end, server side, mobile side, and game side.In order to introduce sentry to `Hertz` on the basis of `Sentry-Go`, we implement the middleware hertzsentry, which provides some unified interfaces to help users get `sentry hub` and report error messages.
+
+### Installation
+
+```go
+go get github.com/getsentry/sentry-go/gin
+```
+
+```go
```
Comment (li-jin-gou) [inline README.md:15]: readme‘s code need to format

File: hertzsentry.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/cloudwego/hertz/pkg/app"
+	"github.com/cloudwego/hertz/pkg/common/adaptor"
+	"github.com/getsentry/sentry-go"
+	"time"
+)
+
+const valuesKey = "sentry-hub"
+
+// Option defines the config for hertz sentry.
+type Option struct {
```
Comment (li-jin-gou) [inline hertzsentry.go]: use `option` instead of `config` 

refer to https://github.com/hertz-contrib/keyauth

File: option.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+type Option struct {
+	F func(o *Options)
+}
+
+// Options defines the config for hertz sentry.
+type Options struct {
+	// RePanic configures whether Sentry should repanic after recovery.
+	// Set to true, if Recover middleware is used.
+	// Optional. Default: false
+	RePanic bool
```
Comment (Skyenought) [inline option.go]: 使用 function options 就把 props 私有化吧

File: option.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+	return options
+}
+
+func (o *Options) Apply(opts []Option) {
+	for _, op := range opts {
+		op.F(o)
+	}
+}
+
+func WithRePanic(rePanic bool) Option {
```
Comment (Skyenought) [inline option.go:64]: 你可以把注释写在配置函数上，这样鼠标指针悬浮在函数的时候就可以直接显示注释，而不是进入 struct 中查看

File: option.go [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+import "time"
+
+// Option set to config unique option
+type Option struct {
+	F func(o *Options)
+}
+
+// Options defines the config for hertz sentry.
+type Options struct {
```
Comment (Skyenought) [inline option.go]: Options -> options

File: option.go [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+	timeout time.Duration
+}
+
+func NewOptions(opts ...Option) options {
+	cfg := options{
+		rePanic:         false,
+		waitForDelivery: false,
+		sendRequest:     false,
+		sendBody:        false,
+		timeout:         2,
```
Comment (li-jin-gou) [inline option.go]: 2 => 2 *  time.Second

File: option.go [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+func WithSendBody(sendBody bool) Option {
+	return Option{F: func(o *options) {
+		o.sendBody = sendBody
+	}}
+}
+
+// WithTimeout configs timeout for the event delivery requests.
+func WithTimeout(timeout time.Duration) Option {
+	return Option{F: func(o *options) {
+		if timeout == 0 {
```
Comment (li-jin-gou) [inline option.go]: 因为已经有默认值了，所以这个地方的设置是多余的

File: hertz_sentry_test.go [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hertzsentry
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudwego/hertz/pkg/app"
```
Comment (li-jin-gou) [inline hertz_sentry_test.go]: import need to fomat

**Review by li-jin-gou — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Review by Skyenought — APPROVED** [review]
LGTM

**Pull request merged**